gef.py

```diff
@@ -11175,13 +11223,17 @@ def connect(self, pid: int) -> bool:
 
     def setup(self) -> bool:
         # setup remote adequately depending on remote or qemu mode
-        if self.in_qemu_user():
+        if self.mode == GefRemoteSessionManager.RemoteMode.QEMU:
```

Comment:
Couldn't these all be done by making thin subclasses that just override the relevant parts?

Reply:
In theory, I'd agree with this approach and I did it that way first, but here the gef.session object is instantiated as a `GefRemoteSessionManager` class, and we don't know yet if it's a gdbserver, rr or just remote. Using subclasses would deport the logic of determining this to the caller which I don't find really elegant so I went this way

Reply:
But don't we not want to set the `remote` part until we call `gef-remote`? So the command itself could select the appropriate subclass based on the args.